test/helpers/position_assertions.cc

```diff
@@ -1129,4 +1130,119 @@ void ApplyCodeActionAssertion::check(const UnorderedMap<std::string, std::shared
     }
 }
 
+SymbolSearchAssertion::SymbolSearchAssertion(string_view filename, unique_ptr<Range> &range, int assertionLine,
+                                             string_view query)
+    : RangeAssertion(filename, range, assertionLine), query(query) {}
+
+shared_ptr<SymbolSearchAssertion> SymbolSearchAssertion::make(string_view filename, unique_ptr<Range> &range,
+                                                              int assertionLine, string_view assertionContents,
+                                                              string_view assertionType) {
+    static const regex contentsRegex(R"(^\s*\"([^\"]+)\"\s*$)");
+
+    smatch matches;
+    string assertionContentsString = string(assertionContents);
+    if (regex_search(assertionContentsString, matches, contentsRegex)) {
+        auto query = matches[1].str();
+        return make_shared<SymbolSearchAssertion>(filename, range, assertionLine, query);
+    }
+
+    ADD_FAILURE_AT(string(filename).c_str(), assertionLine + 1) << fmt::format(
+        "Improperly formatted assertion. Expected 'symbol-search: \"<query>\"'. Found '{}'", assertionContents);
+
+    return nullptr;
```

Comment:
You should probably `Exception::raise` here to throw an exception to abort the test.

Reply:
Running locally, it doesn't seem necessary to raise an exception to abort the test...if I create a malformed line, it segfaults with this output:

    test/testdata/lsp/workspace_symbols_fullname.rb:2: Failure
    Failed
    Improperly formatted assertion. Expected 'symbol-search: "<query>"'. Found '"unclosed query'
    /private/var/tmp/_bazel_mkillianey/556e97dcfeb019a287141d5e3fde2a58/execroot/com_stripe_ruby_typer/bazel-out/darwin-dbg/bin/test/test_LSPTests/testdata/lsp/workspace_symbols_fullname.runfiles/com_stripe_ruby_typer/test/test_LSPTests/testdata/lsp/workspace_symbols_fullname: line 4: 42740 Segmentation fault: 11  test/test_corpus_runner "$@"

Could we push off throwing exceptions to a later PR?  If we decide to take that approach, I'm guessing we also want to change the other `make` methods in this file that do the same thing, like [`ApplyCompletionAssertion::make`](https://github.com/sorbet/sorbet/blob/a7bc630d732b7947cd5214ca5a4796c7b43e4b8a/test/helpers/position_assertions.cc#L970-L988
) and [`ApplyCodeActionAssertion::make`](https://github.com/sorbet/sorbet/blob/a7bc630d732b7947cd5214ca5a4796c7b43e4b8a/test/helpers/position_assertions.cc#L1056-L1073).

Reply:
Ah, I didn't realize you cargo culted. Yeah, it'd be better if things didn't rely on a segfault to stop.